tests/ci_build/Dockerfile.cpu_clang

```diff
@@ -1,21 +0,0 @@
-FROM ubuntu:16.04
-
-COPY install/ubuntu_install_core.sh /install/
```

Comment:
The problem of putting installation bash into in a centralized location, is that change of a single bash will trigger rebuild of entire docker image. 

Having separate bash script is actually preferred and makes it easier to tweak new installations by putting them in the end to take benefit of the cache

Reply:
We plan to make use of a centralized docker repository for all our slaves as part of Auto Scaling, thus reducing the impact. But you're right, a single change will trigger rebuilds for all containers.

Reply:
is there any reason of why we don't want to separate the build scripts? It seems to me only a matter of taste and common docker's build would advocate the other way

Reply:
I agree. I'll check if I might be able to reduce the number of Dockerfiles and thus make centralization obsolete.

Reply:
@tqchen you are right. But separate scripts are more difficult to maintain. In the end we can find a compromise balance. Currently we use a script to create images (install) and a script to build, so changes in build flags don't rebuild images.

Reply:
@tqchen just to be clear, separate scripts are more difficult to maintain / keep consistent

Reply:
This is a bit like maintaining multiple cpp files for logically different functions. It seems to be not that different as long as everything is kept in one folder.

The duplication can happen, however, in the docker file that you wrote, which is like copy a few lines of script invocation vs one line of invocation. 

I personally have quite a lot pain rebuilding docker build every time I tried to add a new dep, especially for those that contain complicated source build like LLVM. So just talking my own experience so that at least we know the possible problem here

Reply:
Thanks a lot for your suggestion. I have made the same experience and I will probably switch back to having every function separately. The runtime scripts (called build_functions.sh) will stay as one file while install_functions.sh will be split into separate files again.